control.lua

```diff
@@ -156,165 +158,205 @@ end
+	surface.set_chunk_generated_status({-1, 0}, defines.chunk_generated_status.entities)
+	surface.set_chunk_generated_status({0, -1}, defines.chunk_generated_status.entities)
+	surface.set_chunk_generated_status({-1, -1}, defines.chunk_generated_status.entities)
+	
+	return surface
 end
 
-local function update_power_settings(factory)
-	if factory.built then
-		local layout = factory.layout
-		-- Inside sender
-		local new_ies = factory.inside_surface.create_entity{
-			name = "factory-power-output-2-" .. factory.transfer_rate,
-			position = {factory.inside_x + layout.inside_energy_x, factory.inside_y + layout.inside_energy_y},
-			force = factory.force
-		}
-		new_ies.destructible = false
-		new_ies.operable = false
-		new_ies.rotatable = false
-		if factory.inside_energy_sender.valid then
-			factory.inside_energy_sender.destroy()
+local function remove_direct_connection(factory)
+	local dc = factory.direct_connection
+	if not dc or not dc.valid then return end
+	
+	for _, pole in pairs(factory.inside_power_poles) do
+		for _, neighbour in pairs(pole.neighbours.copper) do
+			if neighbour == dc then
+				local old = {}
+				for _, neighbour in ipairs(dc.neighbours.copper) do
+					if neighbour ~= pole then old[#old+1] = neighbour end
+				end
+				dc.disconnect_neighbour()
+				for _, neighbour in ipairs(old) do
+					dc.connect_neighbour(neighbour)
+				end
+				factory.direct_connection = nil
+				return
+			end
 		end
-		factory.inside_energy_sender = new_ies
+	end
+end
 
-		-- Inside receiver
-		local new_ier = factory.inside_surface.create_entity{
-			name = "factory-power-input-2-" .. factory.transfer_rate,
-			position = {factory.inside_x + layout.inside_energy_x, factory.inside_y + layout.inside_energy_y},
-			force = factory.force
-		}
-		new_ier.destructible = false
-		new_ier.operable = false
-		new_ier.rotatable = false
-		if factory.inside_energy_receiver.valid then
-			factory.inside_energy_receiver.destroy()
+local function delete_middleman(i)
+	local pole = global.middleman_power_poles[i]
+	if pole == 0 then return end
+	global.middleman_power_poles[i] = i < #global.middleman_power_poles and 0 or nil
+	pole.destroy()
+	for _, factory in pairs(global.factories) do
+		if factory.middleman_id == i then
+			factory.middleman_id = nil
 		end
-		factory.inside_energy_receiver = new_ier
+	end
+end
 
-		-- Outside sender
-		local new_oes = factory.outside_surface.create_entity{
-			name = layout.outside_energy_sender_type .. "-" .. factory.transfer_rate,
-			position = {factory.outside_x, factory.outside_y},
-			force = factory.force
-		}
-		new_oes.destructible = false
-		new_oes.operable = false
-		new_oes.rotatable = false
-		if factory.outside_energy_sender.valid then
-			factory.outside_energy_sender.destroy()
-		end
-		factory.outside_energy_sender = new_oes
+local function cleanup_middlemen()
+	for i, pole in ipairs(global.middleman_power_poles) do
+		if pole ~= 0 and #pole.neighbours.copper<2 then delete_middleman(i) end
```

Comment:
Another strange place, checking that pole ~= 0.
If you use pairs, and clear the pole to nil, then you can just use a normal loop

Also you should be checking if the pole entity is valid